rust/validation/src/feedback.rs

```diff
@@ -162,3 +231,229 @@ pub enum ViolationValidValues {
     #[display("{_0:?}")]
     Str(Vec<String>),
 }
+
+#[derive(Clone, Debug, PartialEq, Serialize, derive_more::From)]
+pub struct ReplacementField(Option<String>);
+impl Display for ReplacementField {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        if let Some(replacement) = &self.0 {
+            write!(f, " Use '{replacement}' instead.")
+        } else {
+            Ok(())
+        }
+    }
+}
+impl From<ReplacementField> for Option<String> {
+    fn from(value: ReplacementField) -> Self {
+        value.0
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize, derive_more::From)]
+pub struct VersionField(Option<String>);
+impl Display for VersionField {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        if let Some(version) = &self.0 {
+            write!(f, " in AVD version {version}")
+        } else {
+            Ok(())
+        }
+    }
+}
+impl From<VersionField> for Option<String> {
+    fn from(value: VersionField) -> Self {
+        value.0
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize, derive_more::From)]
+pub struct UrlField(Option<String>);
+impl Display for UrlField {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        if let Some(url) = &self.0 {
+            write!(f, " See '{url}' for details.")
+        } else {
+            Ok(())
+        }
+    }
+}
+impl From<UrlField> for Option<String> {
+    fn from(value: UrlField) -> Self {
+        value.0
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize, derive_more::Display)]
+#[display("The input data model '{path}' is deprecated and will be removed{version}.{replacement}{url}")]
+pub struct Deprecated {
+    pub path: Path,
+    pub replacement: ReplacementField,
+    pub version: VersionField,
+    pub url: UrlField,
+}
+impl Deprecated {
```

Comment:
following my remark on the tests below, maybe we just make one struct here that does the job for both and honor the `removed` field from `avdschema::base::Deprecation`? Like `DeprecationInfo` or whatever?